Add new C64 SVG Template

Stat-Mat opened the pull request:
This is the updated version of my customized label template for Commodore 64 games which is based on the latest HuCard template. Hope you will consider adding it as an official option to your awesome TapTo Designer project. :)

asturur commented:
yes!

asturur commented:
Do you want to try to make a full PR? or do you want to just commit the file?

Stat-Mat commented:
I was just going to commit the file rather than trying to figure out how to get it added to the app. I can try to do that if you'd prefer. I just thought it may be easiser for you. ;)

asturur commented:
It is your preference.
I ll make a PR tomorrow to add a new sligthly modified gameboy template and i'll use it as reference. Then you either just commit the file or try it if you are curious.
I offered because i do not know if you want to try :)

Stat-Mat commented:
Okay, I have corrected the missing background object in the SVG (I inadvertently deleted this when using the original HuCard template as a basis) and added the new C64 HuCard variant to the cards templates, which I think is all that's required.

asturur commented:
if it works, here you can test it:
https://deploy-preview-16--tapto-designer.netlify.app/

asturur commented:
Seems good to me!

asturur commented:
if the button allow you to merge, and you are happy with results press squash and merge. Ofherwise let me know i ll merge it

Stat-Mat commented:
Okay, great to know about the pre-live environment. Thanks for the tip.

It turns out the existing SVGs use the non-standard CSS style fill:transparent which was being ignored by Inkscape. This explains why this was being drawn as a solid black area obscuring most of the other objects. Anyway, I have now fixed the C64 template to use the correct standards way of setting this (i.e. fill-opacity:0) and it's now working correctly in the prelive build now:

1. New option HuCard (C64) appears in the Card template dropdown at the top
2. When selected, it correctly renders the template and selected game image on the page, just like the existing HuCard template
3. Exporting to PDF and Zip also works fine
4. However, the new PDF [beta] (i.e. PDFKit) does not correctly render the image in the PDF

Is the beta PDF option going to be used or removed, given that the main export to PDF option seems to work fine?

Stat-Mat commented:
**Update:** using the non-standard CSS fill:transparent allows the PDFKit (beta) PDF rendering to work better, but it still doesn't render the Commodore logo correctly. Not really sure what is breaking it, as clearly the SVG is good. The only other option would be to find out from Tim which editor he used and re-create the C64 one again with that instead of Inkscape. It all depends on the need for the PDFKit (beta) export option really.

asturur commented:
the pdf beta code will replace the old one.
The current code covers:
- paths
- rects
and nothing else.

If in your SVG there is something different is not going to work for now.

instead of using fill-opacity i think you can use `fill='none'` that is good enough.

fill:transparent is probably CSS only and the browser and fabricJS are able to render it. But if not SVG standard stuff it shouldn't be there.

asturur commented:
Probably is because you use polygon, but that is on my, i ll add support for polygon to the PDF code
